bin/build-cored-release

```diff
@@ -0,0 +1,57 @@
+#!/bin/sh
+set -e
+
+usage() {
+  echo "Usage: $0 <releaseRef> <outputDir>"
+  echo
+  echo "Envvars:"
+  echo "GOOS:   target OS"
+  echo "GOARCH: target CPU architecture"
+  echo "DATE:   build date; defaults to now"
+  exit 1
+}
+
+trap usage ERR
+
+if [ -z "$1" -o -z "$2" ]; then
+  usage
+fi
+
+releaseRef=$1
+cd $2; outputDir=`pwd` # verify output dir and expand to absolute
+
+echo "Building ref $releaseRef to output directory $outputDir..."
+
+# Clone repository into tempdir and check out specified release ref.
+buildGoPath=`mktemp -d`
+trap "rm -rf $buildGoPath" EXIT
+
+newChain=$buildGoPath/src/chain
+mkdir -p $newChain
+git clone $CHAIN $newChain
+cd $newChain
+git checkout $releaseRef
+
+# Cross-compilation args for cored and corectl
+export GOPATH=$buildGoPath
+export GOOS
+export GOARCH
```

Comment:
Currently, I don't think this script will fail if `GOOS` and `GOARCH` are not set. You can assert them with the following statements:
```
: ${GOOS:?"some error text"}
: ${GOARCH:?"some error text"}
```
I also don't think you need to re-export them since they are passed in as env vars.